cpp/oneapi/dal/data/graph_service_functions.hpp

```diff
@@ -65,16 +66,16 @@ constexpr auto get_edge_count(const G &g) noexcept -> edge_size_type<G> {
 template <typename G>
 constexpr auto get_vertex_degree(const G &g, const vertex_type<G> &vertex) noexcept
     -> vertex_edge_size_type<G> {
-    if (vertex < 0 && vertex >= get_vertex_count_impl(g)) {
-        throw out_of_range("Vertex index should be in [0, num_vert)");
+    if (vertex < 0 || (vertex_size_type<G>)vertex >= get_vertex_count_impl(g)) {
```

Comment:
What is the right type of the exception should be here?

Reply:
out_of_range